utils/staking-miner/src/token.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use ss58_registry::{Token as WrappedToken, TokenAmount};
+
+static mut TOKEN: WrappedToken = TokenRegistry::Dot.into();
```

Comment:
no way around the `static mut` here?

Reply:
The rest of the code needs to be injected with a Token or needs another ambient context implementation. The environmental crate comes to my mind. But since the whole crate is single threaded and there is no risk for reentrancy between Token method calls, I was not too concerned about the implementation.